src/plan.rs

```diff
@@ -45,8 +49,12 @@ impl ExecutionPlan {
 
     /// Currently this can only fail due to argument parsing.
     /// Split into its own function due to lack of try{} blocks on stable Rust.
-    fn apply_arg_inner(&mut self, arg: Arg, value: Option<&OsStr>) -> Result<(), ArgParseErr> {
-        match arg {
+    fn apply_arg_inner(
+        &mut self,
+        signed_arg: SignedArg,
+        value: Option<&OsStr>,
+    ) -> Result<(), ArgParseErr> {
+        match signed_arg.arg {
```

Comment:
Also an existing bug: this should probably match against both the sign and the option name, to avoid allowing something like `+auto-orient`.